src/containers/CreateCommunity.js

```diff
@@ -82,7 +99,20 @@ export class CreateCommunity extends React.Component {
   constructor (props) {
     super(props)
     this.state = {
-      expanded: false
+      expanded: false,
+      generatedSlug: '',
+      editedSlug: false
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { editedSlug, generatedSlug } = this.state
+    if (!editedSlug && get('errors.slugUsed', nextProps)) {
+      const newGeneratedSlug = generateSlug(null, generatedSlug)
+      this.setState({
+        generatedSlug: newGeneratedSlug
+      })
+      this.checkUnique('slug', newGeneratedSlug)
```

Comment:
I think you're incrementing by 2 instead of 1 because this is running twice. Try resetting validation before running checkUnique again.